File: ResortERP.UI/ngApp/Shared/Services/emailsService.js
'use strict';
app.factory('emailsService', ['$http', 'sharedService', function ($http, sharedService) {

    var serviceBase = sharedService.getBaseUrl();

    var emailServiceFactory = {};

    var _get = function (pageNum, pageSize) {
        var myUrl = serviceBase + 'email/get/' + pageNum + '/' + pageSize;
        return $http.get(myUrl).then(function (results) {
            return results;
        });
    }

    var _getByUID = function (userID) {
        var myUrl = serviceBase + 'email/getByUID/' + userID;
        return $http.get(myUrl).then(function (results) {
            return results;
        });
    }

    var _getAll = function () {
        var myUrl = serviceBase + 'email/getAll';
        return $http.get(myUrl).then(function (results) {
            return results;
        });
    }

    var _count = function () {
        var myUrl = serviceBase + 'email/count';
        return $http.get(myUrl).then(function (results) {
            return results;
        });
    }

    var _update = function (entity) {
        var myUrl = serviceBase + 'email/update';
        var data = {};
        data = entity;
        return $http({
            method: 'POST',
            url: myUrl,
            data: JSON.stringify(data)
        }).then(function (results) {
            return results;
        });
    }

    var _delete = function (entity) {
        var myUrl = serviceBase + 'email/delete';
        var data = {};
        data = entity;
        return $http({
            method: 'POST',
            url: myUrl,
            data: JSON.stringify(data)
        }).then(function (results) {
            return results;
        });
    }

    var _insert = function (entity) {
        var myUrl = serviceBase + 'email/insert';
        var data = {};
        data = entity;
        return $http({
            method: 'POST',
            url: myUrl,
            data: JSON.stringify(data)
        }).then(function (results) {
            return results;
        });
    }

    emailServiceFactory.get = _get;
    emailServiceFactory.count = _count;
    emailServiceFactory.insert = _insert;
    emailServiceFactory.update = _update;
    emailServiceFactory.delete = _delete;
    emailServiceFactory.getAll = _getAll;
    emailServiceFactory.getByUID = _getByUID;

    return emailServiceFactory;

}]);

File: ResortERP.Service/Services/EmailsService.cs
using ResortERP.Core;
using ResortERP.Core.VM;
using ResortERP.Repository;
using ResortERP.Service.IServices;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using AutoMapper;

namespace ResortERP.Service.Services
{
    public class EmailsService : IEmailsService, IDisposable
    {
        IGenericRepository<Emails> emailsRepo;
        public EmailsService(IGenericRepository<Emails> EmailsRepo)
        {
            this.emailsRepo = EmailsRepo;
        }

        public Task<int> CountAsync()
        {
            return Task.Run<int>(() =>
            {
                return emailsRepo.CountAsync();
            });
        }

        public bool Delete(EmailsVM entity)
        {
            Emails email = new Emails
            {
                id = entity.id
            };
            emailsRepo.Delete(email, entity.id);
            return true;
        }

        public Task<bool> DeleteAsync(EmailsVM entity)
        {
            return Task.Run<bool>(() =>
            {
                Emails email = new Emails
                {
                    id = entity.id
                };

                emailsRepo.Delete(email, entity.id);
                return true;
            });
        }

        public void Dispose()
        {
            emailsRepo.Dispose();
        }

        public List<EmailsVM> GetAll()
        {
            var q = (from entity in emailsRepo.GetAll() select entity).Select(X => Mapper.Map<Emails, EmailsVM>(X));
            return q.ToList();
        }

        public Task<List<EmailsVM>> GetAllAsync(int pageNum, int pageSize)
        {
            return Task.Run<List<EmailsVM>>(() =>
            {
                int rowCount;
                var q = (from entity in emailsRepo.GetPaged<long>(pageNum, pageSize, p => p.id, false, out rowCount)
                         select entity).Select(X => Mapper.Map<Emails, EmailsVM>(X));
                return q.ToList();
            });
        }

        public Task<List<EmailsVM>> GetByUID(int userID)
        {
            return Task.Run<List<EmailsVM>>(() =>
            {
                var q = emailsRepo.Filter(Y => Y.UID_ID == userID).Select(X => Mapper.Map<Emails, EmailsVM>(X));
                return q.ToList();
            });
        }

        public bool Insert(EmailsVM entity)
        {
            Emails email = Mapper.Map<EmailsVM, Emails>(entity);
            Emails em = emailsRepo.Add(email);
            return em != null ? true : false;
        }

        public Task<bool> InsertAsync(EmailsVM entity)
        {
            return Task.Run<bool>(() =>
            {
                Emails email = Mapper.Map<EmailsVM, Emails>(entity);
                Emails em = emailsRepo.Add(email);
                return em != null ? true : false;
            });
        }

        public bool Update(EmailsVM entity)
        {
            Emails email = Mapper.Map<EmailsVM, Emails>(entity);
            Emails em = emailsRepo.Update(email, email.id);
            return em != null ? true : false;
        }

        public Task<bool> UpdateAsync(EmailsVM entity)
        {
            return Task.Run<bool>(() =>
            {
                Emails email = Mapper.Map<EmailsVM, Emails>(entity);
                Emails em = emailsRepo.Update(email, email.id);
                return em != null ? true : false;
            });
        }
    }
}
